tests/std/tests/P0408R7_efficient_access_to_stringbuf_buffer/test.cpp

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <memory_resource>
+#include <sstream>
+#include <string>
+#include <string_view>
+#include <utility>
+
+#include <test_death.hpp>
+
+using namespace std;
+
+constexpr const char empty_string[] = "";
+constexpr const char small_string[] = "a";
+constexpr const char large_string[] =
+    "This is a long long long long long long long long string to avoid small string optimization.";
+
+template <typename Stream>
+struct test_rvalue {

```

Comment:
why does this need to be a function object instead of just a regular function?

Reply:
Just preference. Do I need to change them to function pointers?

Reply:
not pointers, just write a function template and call it